robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowParcel.java

```diff
@@ -253,15 +254,10 @@ public static String nativeReadString(long nativePtr) {
   }
 
   @Implementation @HiddenApi
-  public static Number nativeCreate() {
-    // todo: yikes, perf:
-    // Pick a native ptr that hasn't been used.
-    long/*ptr*/ nativePtrUsed = 0;
-    while (NATIVE_PTR_TO_PARCEL.containsKey(nativePtrUsed)) {
-      nativePtrUsed++;
-    }
-    NATIVE_PTR_TO_PARCEL.put(nativePtrUsed, new ByteBuffer());
-    return castNativePtr(nativePtrUsed);
+  synchronized public static Number nativeCreate() {
+    long nativePtr = nextNativePtr++;
```

Comment:
oh, I see that you initialized `nextNativePtr` with `1`. Did you mean to use `nativePtr` in the `NATIVE_PTR_TO_PARCEL` and `castNativePtr(nextNativePtr);`?